src/plugins/unified_search/public/query_string_input/text_based_languages_editor/editor_footer.tsx

```diff
@@ -75,7 +82,10 @@ export const EditorFooter = memo(function EditorFooter({
                             text-decoration: underline;
                           }
                         `}
-                        onClick={() => setIsPopoverOpen(!isPopoverOpen)}
+                        onClick={() => {
+                          refreshErrors();
```

Comment:
About this part of the code. Before, the errors were not updating, if the input was changed, but `CMD + Enter` was not pressed. So, if the line of the error was changed, it was pointing to the previous position. To enable the feature of moving to the position of the error on click, it is required to update the errors before showing the errors list.
But here is a problem. Unfortunately, onSubmit is not returning promises and not giving any controls for tracking the update of errors, so, we can see the change of the errors directly in the modal, which is not good. But for now, it is the only way.